test/compiler/inline.jl

```diff
@@ -1259,3 +1259,20 @@ end
 @test fully_eliminated() do
     return maybe_error_int(1)
 end
+
+# basic tests for inlining of `apply_type` in the presence of unmatched type parameters
+f44656(x) = Val{x}()
+
+function g44656()
+    for i = 1:10000
+        Base.donotdelete(Val{i}())
+    end
+end
+
+let srcs = (code_typed1(f44656, (Any,)),
+            code_typed1(g44656))
+    for src in srcs
+        @test count(isnew, src.code) == 1
+        @test count(iscall((src, Core.apply_type), src.code)) == 0
+    end
+end
```

Comment:
We should add more test cases. E.g. since this PR adds a change on the inlining pass for constant prop'ed callsite, we should have a corresponding test case.